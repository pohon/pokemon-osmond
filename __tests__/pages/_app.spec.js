import App from '../../pages/_app'
import Home from '../../pages/index'
import { render, screen, fireEvent } from '@testing-library/react'

describe('_app test', () => {
  
  it('Should render correctly', () => {
    const { container } = render(<App Component={Home} pageProps={{}} />)
    expect(container).toMatchSnapshot()

  });
})