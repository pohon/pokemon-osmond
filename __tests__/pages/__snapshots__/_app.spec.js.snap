// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_app test Should render correctly 1`] = `
<div>
  <header>
    <title>
      Pokémon
    </title>
    <div
      class="relative bg-white"
    >
      <div
        class="max-w-7xl mx-auto"
      >
        <div
          class="relative z-10 pb-8 bg-white  lg:max-w-2xl lg:w-full sm:pb-16 md:pb-20 lg:pb-28 xl:pb-32"
        >
          <svg
            aria-hidden="true"
            class="hidden absolute right-0 inset-y-0 h-full w-48 text-white transform translate-x-1/2 lg:block"
            fill="currentColor"
            preserveAspectRatio="none"
            viewBox="0 0 100 100"
          >
            <polygon
              points="50,0 100,0 50,100 0,100"
            />
          </svg>
          <div>
            <div
              class="relative pt-6 px-4 sm:px-6 lg:px-8"
            >
              <nav
                aria-label="Global"
                class="relative flex items-center justify-between sm:h-10 lg:justify-start"
              >
                <div
                  class="flex items-center flex-grow flex-shrink-0 lg:flex-grow-0"
                >
                  <div
                    class="flex items-center justify-between w-full md:w-auto"
                  >
                    <a
                      href="/"
                    >
                      <span
                        class="sr-only"
                      >
                        Workflow
                      </span>
                      <img
                        class="h-8 w-auto sm:h-10"
                        src="https://upload.wikimedia.org/wikipedia/commons/5/53/Pok%C3%A9_Ball_icon.svg"
                      />
                    </a>
                    <div
                      class="-mr-2 flex items-center md:hidden"
                    >
                      <button
                        aria-expanded="false"
                        class="bg-white rounded-md p-2 inline-flex items-center justify-center text-gray-400 hover:text-gray-500 hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-indigo-500"
                        id="headlessui-popover-button-1"
                        type="button"
                      >
                        <span
                          class="sr-only"
                        >
                          Open main menu
                        </span>
                        <svg
                          aria-hidden="true"
                          class="h-6 w-6"
                          fill="none"
                          stroke="currentColor"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M4 6h16M4 12h16M4 18h16"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="hidden md:block md:ml-10 md:pr-4 md:space-x-8"
                >
                  <a
                    class="font-medium text-gray-500 hover:text-gray-900"
                    href="/"
                  >
                    PokéList
                  </a>
                  <a
                    class="font-medium text-gray-500 hover:text-gray-900"
                    href="/catched"
                  >
                    My Pokémon
                  </a>
                </div>
              </nav>
            </div>
          </div>
          <main
            class="mt-10 mx-auto max-w-7xl px-4 sm:mt-12 sm:px-6 md:mt-16 lg:mt-20 lg:px-8 xl:mt-28"
          >
            <div
              class="sm:text-center lg:text-left"
            >
              <h1
                class="text-4xl tracking-tight font-extrabold text-gray-900 sm:text-5xl md:text-6xl"
              >
                <span
                  class="block xl:inline"
                >
                  See & collect your Pokémon
                </span>
                 
                <span
                  class="block text-indigo-600 xl:inline"
                >
                  now
                </span>
              </h1>
              <p
                class="mt-3 text-base text-gray-500 sm:mt-5 sm:text-lg sm:max-w-xl sm:mx-auto md:mt-5 md:text-xl lg:mx-0"
              >
                Anim aute id magna aliqua ad ad non deserunt sunt. Qui irure qui lorem cupidatat commodo. Elit sunt amet fugiat veniam occaecat fugiat aliqua.
              </p>
              <div
                class="mt-5 sm:mt-8 sm:flex sm:justify-center lg:justify-start"
              >
                <div
                  class="rounded-md shadow"
                >
                  <a
                    class="w-full flex items-center justify-center px-8 py-3 border border-transparent text-base font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-700 md:py-4 md:text-lg md:px-10"
                    href="#pokelist"
                  >
                    See all Pokémon
                  </a>
                </div>
                <div
                  class="mt-3 sm:mt-0 sm:ml-3"
                >
                  <a
                    class="w-full flex items-center justify-center px-8 py-3 border border-transparent text-base font-medium rounded-md text-indigo-700 bg-indigo-100 hover:bg-indigo-200 md:py-4 md:text-lg md:px-10"
                    href="#"
                  >
                    Collected Pokémon
                  </a>
                </div>
              </div>
            </div>
          </main>
        </div>
      </div>
      <div
        class="lg:absolute lg:inset-y-0 lg:right-0 lg:w-1/2"
      >
        <img
          alt=""
          class="h-56 w-full object-cover sm:h-72 md:h-96 lg:w-full lg:h-full"
          src="https://cdn1-production-images-kly.akamaized.net/Mp028XhF_dD1TvoF7k-dQmpG58c=/1280x1280/smart/filters:quality(75):strip_icc():format(webp)/kly-media-production/medias/3581474/original/088575000_1632412791-Poke__mon_Unite.jpeg"
        />
      </div>
    </div>
  </header>
  <main>
    <div
      id="pokelist"
      style="background: url(https://tailwindcss.com/_next/static/media/hero@75.b2469a49.jpg);"
    >
      <div
        class="max-w-2xl mx-auto py-16 px-4 sm:py-24 sm:px-6 lg:max-w-7xl lg:px-8"
      >
        <div
          class="grid grid-cols-1 gap-y-10 sm:grid-cols-2 gap-x-6 lg:grid-cols-3 xl:grid-cols-4 xl:gap-x-8 mb-8"
        >
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
          <a
            class="group"
          >
            <div
              class="animate-pulse w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
            />
            <h3
              class="animate-pulse mt-4 text-sm text-gray-700 capitalize"
            />
            <p
              class="animate-pulsemt-1 text-lg font-medium text-gray-900"
            />
          </a>
        </div>
        <div
          class="animate-pulse bg-white px-4 py-3 flex items-center justify-between border-t border-gray-200 sm:px-6"
          data-testid="listLoader"
        />
      </div>
    </div>
  </main>
</div>
`;
